File: java/pdfclown.samples.cli/src/org/pdfclown/samples/cli/RenderingSample.java
package org.pdfclown.samples.cli;

import java.awt.geom.Dimension2D;
import java.awt.image.BufferedImage;
import java.io.IOException;

import javax.imageio.ImageIO;

import org.pdfclown.documents.Document;
import org.pdfclown.documents.Page;
import org.pdfclown.documents.Pages;
import org.pdfclown.files.File;
import org.pdfclown.tools.Renderer;
import org.pdfclown.util.io.IOUtils;

/**
  This sample demonstrates <b>how to render a PDF page as a raster image</b>.
  <p>Note: rendering is currently in pre-alpha stage; therefore this sample is
  nothing but an initial stub (no assumption to work!).</p>

  @author Stefano Chizzolini (http://www.stefanochizzolini.it)
  @since 0.1.0
  @version 0.1.2.1, 04/08/15
*/
public class RenderingSample
  extends Sample
{
  @Override
  public void run(
    )
  {
    File file = null;
    try
    {
      // 1. Opening the PDF file...
      {
        String filePath = promptFileChoice("Please select a PDF file");
        try
        {file = new File(filePath);}
        catch(Exception e)
        {throw new RuntimeException(filePath + " file access error.",e);}
      }
      Document document = file.getDocument();
      Pages pages = document.getPages();

      // 2. Page rasterization.
      int pageIndex = promptPageChoice("Select the page to render", pages.size());
      Page page = pages.get(pageIndex);
      Dimension2D imageSize = page.getSize();
      Renderer renderer = new Renderer();
      BufferedImage image = renderer.render(page, imageSize);

      // 3. Save the page image!
      try
      {ImageIO.write(image,"jpg",new java.io.File(getOutputPath("ContentRenderingSample.jpg")));}
      catch(IOException e)
      {e.printStackTrace();}
    }
    finally
    {
      // 4. Closing the PDF file...
      IOUtils.closeQuietly(file);
    }
  }
}

File: PdfClown-NetStandard/CliSamples/org/pdfclown/samples/cli/RenderingSample.cs
using org.pdfclown.documents;
using org.pdfclown.files;
using org.pdfclown.tools;

using System;
using System.Collections.Generic;
using System.Drawing;
using System.Drawing.Imaging;

namespace org.pdfclown.samples.cli
{
  /**
    <summary>This sample demonstrates how to render a PDF page as a raster image.<summary>
    <remarks>Note: rendering is currently in pre-alpha stage; therefore this sample is
    nothing but an initial stub (no assumption to work!).</remarks>
  */
  public class RenderingSample
    : Sample
  {
    public override void Run(
      )
    {
      // 1. Opening the PDF file...
      string filePath = PromptFileChoice("Please select a PDF file");
      using(var file = new File(filePath))
      {
        Document document = file.Document;
        Pages pages = document.Pages;
  
        // 2. Page rasterization.
        int pageIndex = PromptPageChoice("Select the page to render", pages.Count);
        Page page = pages[pageIndex];
        SizeF imageSize = page.Size;
        Renderer renderer = new Renderer();
        Image image = renderer.Render(page, imageSize);
  
        // 3. Save the page image!
        image.Save(GetOutputPath("ContentRenderingSample.jpg"), ImageFormat.Jpeg);
      }
    }
  }
}